internal/cmd/test.go

```diff
@@ -0,0 +1,198 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type GithubTags struct {
+	TagName string         `json:"tag_name"`
+	URL     string         `json:"url"`
+	Assets  []GithubAssets `json:"assets"`
+}
+
+type GithubAssets struct {
+	Name               string `json:"name"`
+	BrowserDownloadURL string `json:"browser_download_url"`
+	URL                string `json:"url"`
+}
+
+type ResourcesConfig struct {
+	TasksConfig     []ResourceConfig `yaml:"tasks"`
+	PipelinesConfig []ResourceConfig `yaml:"pipelines"`
+}
+
+type ResourceConfig struct {
+	Name    string `yaml:"name"`
+	Version string `yaml:"version"`
+}
+
+type CatalogConfig struct {
+	Resources ResourcesConfig `yaml:"resources"`
+}
+
+type Catalog struct {
+	Version string        `yaml:"version"`
+	Catalog CatalogConfig `yaml:"catalog"`
+}
+
+type ResourceInfo struct {
+	Source  string
+	Version string
+}
+
+func testNameConflicts(m GitHubMatrixObject) error {
+	tempDirPath, err := os.MkdirTemp("", "example")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDirPath)
+
+	kindSourceMap := make(map[string]map[string][]ResourceInfo)
+	kindSourceMap["tasks"] = make(map[string][]ResourceInfo)
+	kindSourceMap["pipelines"] = make(map[string][]ResourceInfo)
+
+	for _, githubObj := range m.Include {
+		var orgURL string
+		partsOfURL := strings.Split(githubObj.URL, "github.com/")
+
+		if len(partsOfURL) > 1 {
+			orgURL = partsOfURL[1]
+		} else {
+			return fmt.Errorf("incorrect url i.e. url doesn't contain github.com")
+		}
+
+		tagsURL := "https://api.github.com/repos/" + orgURL + "/releases"
+		tagsResp, err := MakeGetRequest(tagsURL)
+		if err != nil {
+			return err
+		}
+		defer tagsResp.Body.Close()
+
+		tagsBody, err := io.ReadAll(tagsResp.Body)
+		if err != nil {
+			return err
+		}
+
+		var releases []GithubTags
+		err = json.Unmarshal(tagsBody, &releases)
+		if err != nil {
+			return err
+		}
+
+		for _, release := range releases {
+			if !strings.Contains(githubObj.IgnoreVersions, release.TagName) {
+				for _, asset := range release.Assets {
+					if asset.Name == "catalog.yaml" {
```

Comment:
Can you use this variable here https://github.com/openshift-pipelines/catalog-cd/blob/main/internal/contract/contract.go#L18 which is `Filename` instead of hard coding ?